File: 13-fibonacci/fibo.exs
# Elixir program to print the first n fibonacci numbers
# Created by M.V.Harish Kumar on 24-12-2023

defmodule Fib do
  def fib(n), do: fib(n, 0, 1)
  def fib(0, a1, _), do: a1
  def fib(n, a1, a2), do: fib(n-1, a1+a2, a1)
end

IO.puts "Fibonacci series printer"
{nterms, _} = IO.gets("Enter the no. of terms: ") |> Integer.parse
IO.puts "The first #{nterms} terms of fibonacci series are:\n"
Enum.each(1..nterms, fn i ->
  IO.puts Fib.fib(i)
end)

File: 13-fibonacci/fibo.rb
# Fibonacci series printer in Ruby
# Created by M.V.Harish Kumar on 01-02-2025

puts "Fibonacci Series Printer", ''
print 'Enter no of terms to print: '
n = gets.to_i
puts "The first #{n} terms of fibonacci series are:"
a,b = 0, 1
n.times do
  puts a
  a, b = b, a+b
end
